sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -407,59 +493,94 @@ private void initializeCryptoClients() {
-     * @throws UnsupportedOperationException if the wrap operation is not supported or configured on the key.
-     * @throws NullPointerException if {@code algorithm} or {@code key} is null.
+     * @param key The key content to be wrapped.
+     * @return A {@link Mono} containing a {@link WrapResult} whose {@link WrapResult#getEncryptedKey() encrypted key}
+     * contains the wrapped key result.
+     * @throws ResourceNotFoundException If the key cannot be found for wrap operation.
+     * @throws UnsupportedOperationException If the wrap operation is not supported or configured on the key.
+     * @throws NullPointerException If {@code algorithm} or {@code key} are {@code null}.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<WrapResult> wrapKey(KeyWrapAlgorithm algorithm, byte[] key) {
+        return wrapKey(algorithm, key, null);
+    }
+
+    /**
+     * Wraps a symmetric key using the configured key. The wrap operation supports wrapping a symmetric key with both
+     * symmetric and asymmetric keys. This operation requires the keys/wrapKey permission.
+     *
+     * <p>The {@link KeyWrapAlgorithm wrap algorithm} indicates the type of algorithm to use for wrapping the specified
+     * key content. Possible values include:
+     * {@link KeyWrapAlgorithm#RSA1_5 RSA1_5}, {@link KeyWrapAlgorithm#RSA_OAEP RSA_OAEP} and {@link
+     * KeyWrapAlgorithm#RSA_OAEP_256 RSA_OAEP_256}.
+     *
+     * Possible values for symmetric keys include: {@link EncryptionAlgorithm#A128KW A128KW},
+     * {@link EncryptionAlgorithm#A192KW A192KW} and {@link EncryptionAlgorithm#A256KW A256KW}.</p>
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Wraps the key content. Subscribes to the call asynchronously and prints out the wrapped key details when a
+     * response has been received.</p>
+     * {@codesnippet com.azure.security.keyvault.keys.cryptography.CryptographyAsyncClient.wrapKey#KeyWrapAlgorithm-byte-CryptographyOptions}
+     *
+     * @param algorithm The encryption algorithm to use for wrapping the key.
+     * @param key The key content to be wrapped.
+     * @param options Optional parameters for the wrap operation.
+     * @return A {@link Mono} containing a {@link WrapResult} whose {@link WrapResult#getEncryptedKey() encrypted key}
+     * contains the wrapped key result.
+     * @throws ResourceNotFoundException If the key cannot be found for wrap operation.
+     * @throws UnsupportedOperationException If the wrap operation is not supported or configured on the key.
+     * @throws NullPointerException If {@code algorithm} or {@code key} are {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<WrapResult> wrapKey(KeyWrapAlgorithm algorithm, byte[] key, CryptographyOptions options) {
         try {
-            return withContext(context -> wrapKey(algorithm, key, context));
+            return withContext(context -> wrapKey(algorithm, key, options, context));
```

Comment:
In addition to what @JonathanGiles said about "options", currently no "options" are needed for AES-KW and this would be premature and confusing. By using the `EncryptOptions` and `DecryptOptions` for `encrypt` and `decrypt`, we avoid this confusion as well.